### DISCUSSION TITLE
Lookup files outside of local directory

### DISCUSSION THREAD

**Discussion by taraspos** [renovatebot/renovate#29822] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub Actions - 37.414.0

### Please tell us more about your question or problem

I'am trying to automate EKS Addon version upgrade with renovate (similar to the issue here https://github.com/renovatebot/renovate/issues/23410#issuecomment-1655947633) by using custom datasource with `file://` location[^1].

[^1]: https://docs.renovatebot.com/modules/datasource/custom/#custom-offline-dependencies

- Running command like:

  ```bash
   aws eks describe-addon-versions --region us-west-2 --kubernetes-version $EKS_VERISON \
      --query "addons[*].{addonName: addonName, releases: addonVersions[*].{version: addonVersion}}"  > eks-addons.json
  ```
  
  Will populate file of following format:
  
  ```json
  [
      {
          "addonName": "vpc-cni",
          "releases": [
              {
                  "version": "v1.18.2-eksbuild.1"
              },
              {
                  "version": "v1.18.1-eksbuild.3"
              },
              ...
          ]
      },
      {
          "addonName": "upwind-security_upwind-operator",
          "releases": [
              {
                  "version": "v0.3.0-eksbuild.1"
              }
          ]
      },
     ...
  ]
  ```

- And setting up Custom Datasource configuration like:

  ```json
    "customDatasources": {
      "eks-addons": {
        "defaultRegistryUrlTemplate": "file://.versions/eks-addons.json",
        "transformTemplates": [ "{'releases': $$[addonName = '{{packageName}}'].releases}" ]
      }
    },
  ```

This setup works perfectly fine with `platform=local`. However when I'm trying to run this as github action I'm facing one of following problems:
1. When renovatebot clones the repo, it removes the `.versions/eks-addons.json` file that was created in the previous step and mounted into renovatebot container.
2. If I change file location to `file:///tmp/eks-addons.json`, `file:///tmp/renovate/eks-addons.json` or any other location file is ignored with `Preventing access to file outside allowed directory` error because only `localDir` is allowed:
    - https://github.com/renovatebot/renovate/blob/b6d2c4bb32e52a2d8a5d7fe06b89297a9bb3d5b6/lib/util/fs/util.ts#L6-L15

Is there any way how this can be worked around?

**Comment by viceice** [comment]
there is no way, the file needs to be inside the repo for security reasons.

a workaround is to host that file on a simple webserver and fetch it from there.

**Reply by taraspos** [reply]
I tried to do so by starting webserver as github service container, however renovate can't communicate with it, because they are on different networks:
- https://github.com/renovatebot/github-action/issues/852

**Reply by taraspos** [reply]
Hey @viceice!
Would it possible to release new version of github-action with `docker-network` input?
- https://github.com/renovatebot/github-action/pull/853

**Reply by viceice** [reply]
released, was a CI issue.

**Comment by rarkins** [comment, marked as answer]
I'm happy if someone wants to "Suggest an idea" discussion proposing a new global config option "allowNonLocalAccess".
